website/src/content/development/forms.mdx

```diff
@@ -0,0 +1,540 @@
+    onSubmit: async (formikValues) => {
+      alert(`key: ${formikValues.key}`);
+      // Do something async
+    },
+  });
+
+  return (
+    <form onSubmit={formik.handleSubmit}>
+      <TextField
+        name="key"
+        title="Key"
+        isRequired={true}
+        value={formik.values.key}
+        errors={formik.errors.key}
+        touched={formik.touched.key}
+        onChange={formik.handleChange}
+        onBlur={formik.handleBlur}
+      />
+
+      <PrimaryButton
+        type="submit"
+        label="Submit"
+        onClick={formik.handleSubmit}
+        isDisabled={formik.isSubmitting}
+      />
+    </form>
+  );
+}
+```
+
+# Field and input components
+
+The commercetools UIKit library provides several UI components for working with forms. Most of the time you want to use the field components as they provide all the recommended features for rendering form elements.
+
+![Form field example](/images/form-field.png 'Form field example')
+
+A field component consists of an input element wrapped with other field elements such as label, description, error message, hint, badge, etc.
+
+Depending on the use case, you might want to use a date field, or a text field, or a select field. In the UIKit library we provide a variety of components to cover different use cases. For more information about field and input components check their related packages in the [commercetools/ui-kit](https://github.com/commercetools/ui-kit) repository.
+
+<Info>
+
+All UIKit field components have a related input component, for example `<TextField>` -> `<TextInput>`.
+
+</Info>
+
+## Accessibility support
+
+All field and input components have built-in support for accessibility features such as aria labels, keyboard navigation and focus, error messaging.
+
+<Info>
+
+Testing these field components with [React Testing Library](https://github.com/testing-library/react-testing-library) can be easily done using selectors such as `*ByLabelText`, `*ByRole`, etc.
+
+</Info>
+
+# Form validation
+
+One of the important parts of forms is validation. It's important to validate things like required fields, or to check that the value is a valid URL, or even perform asynchronous checks against an API.
+
+For that, Formik allows you to implement a [`validate` function](https://formik.org/docs/api/formik#validate-values-values--formikerrorsvalues--promiseany) that should return an object of errors.
+
+```ts
+type FieldErrors = Record<string, boolean>;
+type FormikErrors = Record<string, FieldErrors>;
+
+declare const validate = (values: FormikValues) => FormikErrors;
+```
+
+Specifically, the `FormikErrors` object should contain properties as the name of the form fields with values being objects with error keys. Let's take a look at an example:
+
+```js title="validate.js"
+import TextInput from '@commercetools-uikit/text-input';
+
+const validate = (values) => {
+  const errors = {};
+
+  if (TextInput.isEmpty(values.key)) {
+    errors.key = { missing: true };
```

Comment:
Nit: I would not use `key` as an example field. It might easily be confusing to reference from text as `The object's key" and "the key field" mean different things. Maybe `sku` or something e-comm specific is easier for readers to follow along with.

Reply:
Yeah, I had `name` initially, but Channel names are localized and I felt like the example would be too complex. Maybe it's good enough, as localized strings are common types in our API.